src/ws/mod.rs

```diff
@@ -96,6 +96,29 @@ where
     WebSocketUpgrade {
         callback,
         config: WebSocketConfig::default(),
+        protocols: &[],
+        _request_body: PhantomData,
+    }
+}
+
+/// Create a new [`WebSocketUpgrade`] service that will call the closure with
+/// each connection.
+///
+/// `protocols` is a list of known protocols. If the protocol name specified by `Sec-WebSocket-Protocol` header
+/// to match any of them, the upgrade response will include `Sec-WebSocket-Protocol` header and return the protocol name.
+///
+/// See the [module docs](crate::ws) for more details.
+pub fn ws_with_protocols<F, B, T>(
```

Comment:
I think we should add this as a builder method on `WebSocketUpgrade` similar to [the other methods](https://docs.rs/axum/0.1.2/axum/ws/struct.WebSocketUpgrade.html).

Reply:
I think we should also include a code example. Could just use the graphql protocol name.

Reply:
You are right, I didn't keep the API style uniform.